src/zos-io.cc

```diff
@@ -1141,51 +1141,72 @@ ssize_t getline(char **lineptr, size_t *n, FILE *stream) {
     return pos;
 }
 
-ssize_t getdelim(char **lineptr, size_t *n, int delimiter, FILE *stream) {
-    size_t pos;
-    int c;
+ssize_t getdelim(char **lineptr, size_t *n, int delimiter, FILE *fp) {
+    ssize_t result;
+    size_t cur_len = 0;
 
-    if (lineptr == NULL || n == NULL || stream == NULL) {
+    if (lineptr == NULL || n == NULL || fp == NULL) {
         errno = EINVAL;
         return -1;
     }
 
-    if (*lineptr == NULL) {
-        *n = 128; // Start with a reasonable buffer size
-        *lineptr = (char *)malloc(*n);
+    if (*lineptr == NULL || *n == 0) {
+        *n = 120;
+        *lineptr = (char *)realloc(*lineptr, *n);
```

Comment:
Calling `realloc()` this was is incorrect.  If the memory allocation fails, then the original value of `lineptr` is lost, and the memory it pointed to is leaked.  `cppcheck` complains about this, and suggests you use a temporary pointer to hold the return value, and then assign it to `lineptr` after validation.

https://stackoverflow.com/questions/21006707/proper-usage-of-realloc